File: bench/times/times.rs
fn add_five(x: i64) -> i64 {
    x + 5
}

fn times(n: i64, f: fn(i64) -> i64, x: i64) -> i64 {
    let mut val = x;
    for _ in (0 .. n) {
        val = f(val)
    }
    val
}

fn main() {
    println!("{}", times(100000, add_five, 0));
}

File: bench/times/times.java
@FunctionalInterface
interface Func {
    public long run(long x);
}

public class times {
    public static long add_five(long x) {
        return x + 5;
    }

    public static long times(long n, Func f, long x) {
        long val = x;
        for(int i = 0; i < n; ++i) {
            val = f.run(val);
        }

        return val;
    }

    public static void main(String[] args) {
        System.out.printf("%d\n", times(1000, times::add_five, 0));
    }
}
